### ISSUE TITLE
ConvolutionFitSequential algorithm

### ISSUE THREAD

**Issue by DanNixon** [mantidproject/mantid#12419]
This issue was originally [TRAC 11581](http://trac.mantidproject.org/mantid/ticket/11581)

This should create a new algorithm `ConvolutionFitSequential` which does the processing of the `confitSeq` function in `IndirectDataAnalysis.py`, with the exception of plotting and saving.

Fit type and background type can be determined by inspecting the fit function.

There should be some validation on the fit function to ensure it is valid for convolution fitting, i.e. it should contain the `Convolution` and `Resolution` fit functions, other than that the scope for checking this is limited as IIRC you can't get a list of functions per category from the FunctionFactory in Python.

The `convertParametersToWorkspace` function in `IndirectCommon.py` should be converted to an algorithm (named something like `ProcessIndirectFitParameters`), this function is used in other places so leave the original function intact, switching other use of the function to the new algorithm can be a separate issue. (This issue has since been created #13336)

Saving can be added to the algorithm queue in the UI using `IndirectTab::addSaveWorkspaceToQueue()`.

Plotting can be done in the callback from the algorithm runner using `IndirectTab::plot*()`.

Both new algorithms should go into the Workflow/MIDAS category.

This issue is blocked by:
- [x] #13300 
- [x] #13336

**Comment by DanNixon** [comment]
There is too much done here for it to go untested in the UI.

**Comment by ElliotAOram** [comment]
Work postponed on this issue until the completion of #13300 as it is required for fit function validation

**Comment by ElliotAOram** [comment]
Completion of #13336 required. In order to run the convertParametersToWorkspace algorithm call.

**Comment by ElliotAOram** [comment]
#13336 is highly related to this issue. It is probably easiest to complete this issue with a python call to convertParamsToWorkspace then after merging, test #13336 within this algorithm.
